spec/system/distribution_system_spec.rb

```diff
@@ -26,15 +26,15 @@
   context "When creating a new distribution manually" do
     it "Allows a distribution to be created" do
       # update items to check for inactive ones
-      @storage_location.items.first.update!(name: 'Inactive Item', active: false)
+      # @storage_location.items.first.update!(name: 'Inactive Item', active: false)
 
       visit @url_prefix + "/distributions/new"
 
       select @partner.name, from: "Partner"
       select @storage_location.name, from: "From storage location"
       choose "Pick up"
 
-      expect(page).not_to have_content('Inactive')
+      # expect(page).not_to have_content('Inactive')
```

Comment:
@dorner @awwaiid @scooter-dangle

I'am pinging you all to get your opinion on this since we worked together last week to uncover the issue with the LineItem list being incorrect. I noticed that this spec started failing when I applied this fix and started including an "Inactive" item and I wanted to get your thoughts. I was thinking that we should try our best to avoid hiding/showing and changing the contents of this dropdown despite being active or having quantities. 

I argue that not showing an "Inactive" item can lead to a race condition. Say a partner requests an item and then the bank marks it as inactive. When the bank tries to fulfill that request, that item that was marked as inactive is going to fall back to some other incorrect item on that list. It seems like removing the item from the list is going to cause trouble.

If our goal is to make the list smaller and easier to manage, then I suggest we apply a different tactic like using autocomplete or typeahead to improve the UX for big lists.

Looking forward to your thoughts!

Reply:
Falling back to some other item is flat out wrong and should never happen. If an item is marked as inactive, we probably should remove it from any active requests and send some kind of notification to the requester indicating that one of their items is no longer active.

If the request is already complete, we shouldn't be making any changes to it - we need to ensure that complete requests still show up exactly as they were when they were made. This might tie into the historical pricing issue (#3053 ).

The other option is to disallow setting any item inactive if there are pending requests, which seems like it moves the responsibility in the wrong direction.

Reply:
@dorner I totally agree about not falling back. In fact, @scooter-dangle and I talked during the office hour that things "auto-magically" disappearing is really unideal and is the source of a lot of issues. 

> If the request is already complete, we shouldn't be making any changes to it - we need to ensure that complete requests still show up exactly as they were when they were made. This might tie into the historical pricing issue (https://github.com/rubyforgood/human-essentials/issues/3053 ).

So taking a step back, I think attempting to remove and alter the list of items accessible in the dropdown is not necessary. The original problem was that the list would be too long for users and removing items was the solution we arose with. However, we believe we can use UI/UX solution to address this problem, using a more improved dropdown like https://tom-select.js.org/ which we can use the typeahead & move the inactive items (or label them) to the bottom of the list.

I'll be reaching out to stakeholders via slack to get their feedback on using this instead of the basic dropdown we have.